lib/parameter_parser/search_parameter_parser.rb

```diff
@@ -215,6 +215,13 @@ def ==(other)
     def valid?
       errors.empty?
     end
+
+    def associated_document_type
+      case field_name
```

Comment:
if you `require "active_support/core_ext/string"` you can `field_name.singularize if field_name == "organisations" `